src/scheduler/MpiWorld.cpp

```diff
@@ -64,14 +64,25 @@ std::shared_ptr<state::StateKeyValue> MpiWorld::getRankHostState(int rank)
     return state.getKV(user, stateKey, MPI_HOST_STATE_LEN);
 }
 
+int MpiWorld::getMpiThreadPoolSize()
+{
+    int usableCores = faabric::util::getUsableCores();
+    int worldSize = size;
+
+    // Note - this may overprovision threads if worldSize > usableCores, and
+    // worldSize % usableCores != 0
+    return std::min<int>(worldSize, usableCores);
```

Comment:
Could we log a warning here if it is overprovisioning? Sometimes it's good to grep the logs for warnings to see what might be going wrong.